clients/src/main/java/org/apache/kafka/clients/consumer/internals/MembershipManager.java

```diff
@@ -75,6 +75,11 @@ public interface MembershipManager {
      */
     void onHeartbeatResponseReceived(ConsumerGroupHeartbeatResponseData response);
 
+    /**
+     * Notify the member that an error heartbeat response was received.
+     */
+    void onHeartbeatError();
```

Comment:
I wonder if it would make sense to rename `onHeartbeatResponseReceived` in this change, since an error response is also a response that we receive -- it should be clear that it's only called in the successful case. Maybe we could call the two functions `onHeartbeatSuccess` and `onHeartbeatFailure`, going along with the convention in `RequestFutureListener` etc.pp.?